src/libraries/System.Private.Xml/src/System/Xml/Serialization/SourceInfo.cs

```diff
@@ -231,25 +232,25 @@ private void ConvertNullableValue(Type nullableType, Type targetType)
             return source.Source;
         }
 
-        public static bool operator !=(SourceInfo a, SourceInfo b)
+        public static bool operator !=(SourceInfo? a, SourceInfo? b)
         {
-            if ((object)a != null)
+            if ((object?)a != null)
```

Comment:
Does this change the code gen here? Also, I'm not sure if `SourceInfo` is a value type. If it's not, maybe we can just remove the cast to `object`??

Reply:
No, it should not change the code gen (also I checked IL diff for this PR as part of the compiler bug check and this has not showed up). object and SourceInfo are both reference types. I prefer to keep the cast in case there is some `!=` operator implemented which this would bypass - it makes it easier on every dimension to keep code as is and also less chance of accidentally breaking anything. Let's leave optimizations and improvements as separate PR as those PRs are large and having any product problems here could be easily lost